composeApp/src/commonMain/kotlin/org/ooni/probe/data/repositories/PreferenceRepository.kt

```diff
@@ -0,0 +1,179 @@
+package org.ooni.probe.data.repositories
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+
+class PreferenceRepository(
+    private val dataStore: DataStore<Preferences>,
+) {
+    /**
+     * This function is used to resolve the preference key for a given test.
+     * The preference key is the name of the test prefixed with the [prefix] of the descriptor
+     * and suffixed with "_autorun" if [autoRun] is true.
+     *
+     * Example: the preference key for the test "web_connectivity" in the
+     * descriptor "websites" is "websites_web_connectivity".
+     * If [autoRun] is true, the preference key is "websites_web_connectivity_autorun".
+     *
+     * @param name The name of the preference
+     * @param prefix The prefix of the preference
+     * @param autoRun If the preference is for auto run
+     * @return The preference key
+     */
+    fun getPreferenceKey(
+        name: String,
+        prefix: String? = null,
+        autoRun: Boolean = false,
+    ): String {
+        return "${prefix?.let { "${it}_" } ?: ""}$name${if (autoRun) "_autorun" else ""}"
+    }
+
+    fun preferenceKeyFromSettingsKey(
```

Comment:
If `getValueByKey` and `setValueByKey` would use `SettingsKey` as well, this method could be private.